src/bin/pg_autoctl/cli_enable_disable.c

```diff
@@ -402,3 +432,593 @@ cli_disable_maintenance(int argc, char **argv)
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	/*
+	 * If any --ssl-* option is provided, either we have a root ca file and a
+	 * server.key and a server.crt or none of them. Any other combo is a
+	 * mistake.
+	 */
+	if (sslCommandLineOptions == SSL_CLI_UNKNOWN)
+	{
+		log_fatal("Explicit SSL choice is required: please use either "
+				  "--ssl-self-signed or provide your certificates "
+				  "using --ssl-ca-file, --ssl-crl-file, "
+				  "--server-key, and --server-crt (or use --no-ssl if you "
+				  "are very sure that you do not want encrypted traffic)");
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!pgsetup_validate_ssl_settings(&(options.pgSetup)))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	/* publish our option parsing in the global variable */
+	keeperOptions = options;
+
+	return optind;
+}
+
+
+/*
+ * cli_enable_ssl enables SSL setup on this node.
+ *
+ *  - edit our Postgres configuration with the given SSL files and options
+ *  - when run on a keeper, checks that the monitor accepts ssl connections
+ *  - edits our configuration at pg_autoctl.conf
+ */
+static void
+cli_enable_ssl(int argc, char **argv)
+{
+	KeeperConfig options = keeperOptions;
+
+	bool missingPgdataIsOk = true;
+	bool pgIsNotRunningIsOk = true;
+	bool monitorDisabledIsOk = true;
+
+	switch (ProbeConfigurationFileRole(options.pathnames.config))
+	{
+		case PG_AUTOCTL_ROLE_MONITOR:
+		{
+			MonitorConfig mconfig = { 0 };
+			PostgresSetup *pgSetup = &(mconfig.pgSetup);
+			LocalPostgresServer postgres = { 0 };
+
+			if (!monitor_config_init_from_pgsetup(&mconfig,
+												  &options.pgSetup,
+												  missingPgdataIsOk,
+												  pgIsNotRunningIsOk))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* now override current on-file settings with CLI ssl options */
+			pgSetup->ssl = options.pgSetup.ssl;
+
+			local_postgres_init(&postgres, pgSetup);
+
+			/* update the Postgres SSL setup and maybe create the certificate */
+			if (!update_ssl_configuration(&postgres, mconfig.nodename))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_INTERNAL_ERROR);
+			}
+
+			/* update the monitor's configuration to use SSL */
+			if (!monitor_config_write_file(&mconfig))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* display a nice summary to our users */
+			log_info("Successfully enabled new SSL configuration:");
+			log_info("  SSL is now %s",
+					 pgSetup->ssl.active ? "active" : "disabled");
+
+			if (pgSetup->ssl.createSelfSignedCert)
+			{
+				log_info("  Self-Signed certificates have been created and "
+						 "deployed in Postgres configuration settings "
+						 "ssl_key_file and ssl_cert_file");
+			}
+
+			log_info("  Health-check connections to Postgres nodes now "
+					 "use sslmode=%s",
+					 pgsetup_sslmode_to_string(pgSetup->ssl.sslMode));
+
```

Comment:
Healt checks never use SSL at the moment, so let's remove this log as it's simply incorrect. We can add it again if we change this.

```suggestion
```